### ISSUE TITLE
Ensure Actions are registered declaratively rather than programatically

### ISSUE THREAD

**Issue by mkondratek** [VirtusLab/git-machete-intellij-plugin#220]
_Originally posted by @mkondratek in https://github.com/VirtusLab/git-machete-intellij-plugin/pull/217#discussion_r415101157_:

So there are two ways to register actions. Programatically (like in `com.virtuslab.gitmachete.frontend.ui.root.GitMachetePanel#createGitMacheteVerticalToolbar`) or declaratively in `plugin.xml`. 

Reference: 
https://www.jetbrains.org/intellij/sdk/docs/basics/action_system.html

If it comes to `GitMachetePanel#createGitMacheteVerticalToolbar`: the refresh action is (and was) implemented here. The slide-out action requires to call it after it is done. Current implementation and (probably too strict?) relation between the panel and refresh disallow registering the refresh action via plugin.xml. That is why I used in-code registering.

Actually, to avoid the re-registered action issue that happened here:

<details><summary>stack trace</summary>
<pre>
2020-04-24 17:31:09,286 [  44157]  ERROR - nSystem.impl.ActionManagerImpl - action with the ID "GitMachete.RefreshAction" was already registered. Action being registered is Refresh Status (Refresh status); Registered action is Refresh Status (Refresh status) 
java.lang.Throwable: action with the ID "GitMachete.RefreshAction" was already registered. Action being registered is Refresh Status (Refresh status); Registered action is Refresh Status (Refresh status)
        at com.intellij.openapi.diagnostic.Logger.error(Logger.java:146)
        at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.reportActionError(ActionManagerImpl.java:400)
        at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.reportActionError(ActionManagerImpl.java:389)
        at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.registerChameleon(ActionManagerImpl.java:1244)
        at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.addToMap(ActionManagerImpl.java:1217)
        at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.registerAction(ActionManagerImpl.java:1188)
        at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.registerAction(ActionManagerImpl.java:1180)
        at com.intellij.openapi.actionSystem.impl.ActionManagerImpl.registerAction(ActionManagerImpl.java:1255)
        at com.virtuslab.gitmachete.frontend.ui.root.GitMachetePanel.createGitMacheteVerticalToolbar(GitMachetePanel.java:65)
        at com.virtuslab.gitmachete.frontend.ui.root.GitMachetePanel.<init>(GitMachetePanel.java:54)
        at com.virtuslab.gitmachete.frontend.ui.root.GitMacheteContentProvider.initContent(GitMacheteContentProvider.java:19)
        at com.intellij.openapi.vcs.changes.ui.ChangesViewContentProvider.initTabContent(ChangesViewContentProvider.java:25)
        at com.intellij.openapi.vcs.changes.ui.ChangesViewContentManager$ContentProvidersListener.selectionChanged(ChangesViewContentManager.kt:169)
        at jdk.internal.reflect.GeneratedMethodAccessor288.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at com.intellij.util.EventDispatcher.dispatchVoidMethod(EventDispatcher.java:127)
        at com.intellij.util.EventDispatcher.access$000(EventDispatcher.java:21)
        at com.intellij.util.EventDispatcher$1.invoke(EventDispatcher.java:85)
        at com.sun.proxy.$Proxy84.selectionChanged(Unknown Source)
        at com.intellij.ui.content.impl.ContentManagerImpl.fireSelectionChanged(ContentManagerImpl.java:579)
        at com.intellij.ui.content.impl.ContentManagerImpl.addSelectedContent(ContentManagerImpl.java:397)
        at com.intellij.ui.content.impl.ContentManagerImpl$1.run(ContentManagerImpl.java:487)
        at com.intellij.ui.content.impl.ContentManagerImpl.lambda$setSelectedContent$4(ContentManagerImpl.java:500)
        at com.intellij.openapi.util.ActionCallback$ExecutedExecutionCallback.doWhenExecuted(ActionCallback.java:177)
        at com.intellij.openapi.util.ActionCallback.doWhenDone(ActionCallback.java:111)
        at com.intellij.openapi.util.ActionCallback.doWhenProcessed(ActionCallback.java:129)
        at com.intellij.ui.content.impl.ContentManagerImpl.setSelectedContent(ContentManagerImpl.java:500)
        at com.intellij.ui.content.impl.ContentManagerImpl.setSelectedContentCB(ContentManagerImpl.java:454)
        at com.intellij.ui.content.impl.ContentManagerImpl.setSelectedContentCB(ContentManagerImpl.java:443)
        at com.intellij.ui.content.impl.ContentManagerImpl.setSelectedContent(ContentManagerImpl.java:437)
        at com.intellij.openapi.wm.impl.content.ContentTabLabel.selectContent(ContentTabLabel.java:243)
        at com.intellij.openapi.wm.impl.content.ContentTabLabel$2.execute(ContentTabLabel.java:138)
        at com.intellij.util.ui.BaseButtonBehavior$MyMouseListener.execute(BaseButtonBehavior.java:173)
        at com.intellij.util.ui.BaseButtonBehavior$MyMouseListener.mouseReleased(BaseButtonBehavior.java:156)
        at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:297)
        at java.desktop/java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:297)
        at java.desktop/java.awt.Component.processMouseEvent(Component.java:6651)
        at java.desktop/javax.swing.JComponent.processMouseEvent(JComponent.java:3345)
        at java.desktop/java.awt.Component.processEvent(Component.java:6416)
        at java.desktop/java.awt.Container.processEvent(Container.java:2263)
        at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5026)
        at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2321)
        at java.desktop/java.awt.Component.dispatchEvent(Component.java:4858)
        at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4918)
        at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4547)
        at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4488)
        at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2307)
        at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2773)
        at java.desktop/java.awt.Component.dispatchEvent(Component.java:4858)
        at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:778)
        at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
        at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
        at java.base/java.security.AccessController.doPrivileged(Native Method)
        at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
        at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:95)
        at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:751)
        at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:749)
        at java.base/java.security.AccessController.doPrivileged(Native Method)
        at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
        at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:748)
        at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:974)
        at com.intellij.ide.IdeEventQueue.dispatchMouseEvent(IdeEventQueue.java:912)
        at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:844)
        at com.intellij.ide.IdeEventQueue.lambda$null$8(IdeEventQueue.java:449)
        at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:739)
        at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$9(IdeEventQueue.java:448)
        at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:831)
        at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:502)
        at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
        at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
        at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
        at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
        at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
        at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
2020-04-24 17:31:09,301 [  44172]  ERROR - nSystem.impl.ActionManagerImpl - IntelliJ IDEA 2020.1  Build #IC-201.6668.121 
2020-04-24 17:31:09,301 [  44172]  ERROR - nSystem.impl.ActionManagerImpl - JDK: 11.0.6; VM: OpenJDK 64-Bit Server VM; Vendor: JetBrains s.r.o 
2020-04-24 17:31:09,301 [  44172]  ERROR - nSystem.impl.ActionManagerImpl - OS: Linux 
2020-04-24 17:31:09,301 [  44172]  ERROR - nSystem.impl.ActionManagerImpl - Plugin to blame: Git Machete version: 0.0.84-SNAPSHOT
</pre>
</details>
and for the sake of consistency, we could consider moving all actions to `plugin.xml`.

## Scope of the task:

- provide checkout action consistency to the rest of actions within their package
- investigate the opportunities that `plugin.xml` provides to register and emplace actions in UI
- perform all required action registrations in `plugin.xml` (e.g. refresh action)
- probably simplify code panel/toolbars creation using `plugin.xml` benefits

**Comment by mkondratek** [comment]
While we are here I suggest handling the `selectedBranch` presence like in `RebaseSelectedBranchOntoParentAction`. So do it in `update` method. In the case of the absence make action invisible. 

Actually there is an interesting case here cause we do not check whether the selected branch belongs to the git machete repository. Is it a bug or a feature? :smile: If it is a bug then add this check to update too. 

If this PR is gonna be merged after #217 then `com.virtuslab.gitmachete.frontend.actions.ActionUtils#getSelectedMacheteBranch` might be useful.

_Originally posted by @mkondratek in https://github.com/_render_node/MDExOlB1bGxSZXF1ZXN0NDA2OTA3NjE4/timeline/more_items_

**Comment by PawelLipski** [comment]
> While we are here I suggest handling the selectedBranch presence like in RebaseSelectedBranchOntoParentAction. So do it in update method. In the case of the absence make action invisible.

Yeah, it seems we're making too many assumptions about certain `DataKey`s being present :/ I'm gonna fix that on one of the PRs

**Comment by PawelLipski** [comment]
> Actually there is an interesting case here cause we do not check whether the selected branch belongs to the git machete repository. 

I'm gonna address this point on a separate PR (not #238)
